diff --git a/node_modules/d3-hexbin/src/hexbin.js b/node_modules/d3-hexbin/src/hexbin.js
index c898b7f..b0ac323 100644
--- a/node_modules/d3-hexbin/src/hexbin.js
+++ b/node_modules/d3-hexbin/src/hexbin.js
@@ -44,11 +44,15 @@ export default function() {
       }
 
       var id = pi + "-" + pj, bin = binsById[id];
-      if (bin) bin.push(point);
+      if (bin) {
+        bin.push(point);
+        bin.data.push(points[i]);
+      }
       else {
         bins.push(bin = binsById[id] = [point]);
         bin.x = (pi + (pj & 1) / 2) * dx;
         bin.y = pj * dy;
+        bin.data = [points[i]];
       }
     }
 
